pandas/tests/series/indexing/test_numeric.py

```diff
@@ -83,19 +83,19 @@ def test_setitem_float_labels():
 
 
 def test_slice_float_get_set(datetime_series):
-    msg = (
-        "cannot do slice indexing on DatetimeIndex with these indexers "
-        r"\[{key}\] of type float"
+    msg = lambda key: (
+        f"cannot do slice indexing on DatetimeIndex with these indexers "
```

Comment:
i'm new to github so i didn't understand your response.
what do you mean by "run black to get linting green"? what does 'run black' mean?
also,may i ask why lambda does not suit here? it's seems funny to add a function just for this,in my opinion. what could be the downside of using lambda expression here?

thank you for your response.

Reply:
thank you for your response.

if i remove the f in the f-string the code doesn't compile.
may i ask why def is preferred for this? i think that lambda suits better because it's a short statement.
also, i published my solution in the thread but didn't get any response

Reply:
Using def vs. lambda is partly a stylistic thing, with the idea being that lambdas are intended as _anonymous_ functions, so naming them somewhat misses the point.

Did you remove the f from all strings? You only need to remove it from the strings for which it has no effect, i.e., when there is nothing to evaluate (e.g., line 87).